scripts/ReportData.js

```diff
@@ -27,27 +30,21 @@ function getPropsAsync(dbId, propNameStr, pieData) {
     // based on some Property Name (eg "Material")
 function getReportDataByPropName(propNameStr, pieOpts, callbackFunc) {
     
-    var leafNodes = getLeafNodeObjs();
-    //console.log("\nLEAF NODES: " + leafNodes.length);   // NOTE: Leaf nodes always is correct
+    getLeafNodeObjs(function(leafNodes) {
+
+        //console.log("\nLEAF NODES: " + leafNodes.length);   // NOTE: Leaf nodes always is correct
+        var allDone = leafNodes.length;
 
         // for each leaf node, find out what Material bucket this object goes in
-    var promises = [];
-    function createPromise() {
-        var p = $.Deferred();
-        promises.push(p);
-        return p.resolve; // Doesn't create an anonymous function
-    };
-    $.each(leafNodes, function(num, dbId) {
-        getPropsAsync(dbId, propNameStr, pieOpts.data);
-    });
-        // set a timer so the array gets loaded correctly
-    //window.setTimeout(function(){$.when.apply($, promises).then(allDone(pieData));},100);
-    window.setTimeout(function() {
-        $.when.apply($, promises).then(function () {
-            if (callbackFunc)
-                callbackFunc(pieOpts);
+        $.each(leafNodes, function(num, dbId) {
+            getPropsAsync(dbId, propNameStr, pieOpts.data, function() {
+                allDone--;
```

Comment:
That's probably the key issue -- you can only call the callback once all the getProps calls are completed. So that's why I added this counter.